tests/v2/test_waku_rln_relay_onchain.nim

```diff
@@ -0,0 +1,263 @@
+
+  return contractAddress
+
+procSuite "Waku-rln-relay":
+  asyncTest "event subscription":
+    # preparation ------------------------------
+    debug "ethereum client address", ETH_CLIENT
+    let contractAddress = await uploadRLNContract(ETH_CLIENT)
+    # connect to the eth client
+    let web3 = await newWeb3(ETH_CLIENT)
+    debug "web3 connected to", ETH_CLIENT
+
+    # fetch the list of registered accounts
+    let accounts = await web3.provider.eth_accounts()
+    web3.defaultAccount = accounts[1]
+    debug "contract deployer account address ",
+        defaultAccount = web3.defaultAccount
+
+    # prepare a contract sender to interact with it
+    var contractObj = web3.contractSender(MembershipContract,
+        contractAddress) # creates a Sender object with a web3 field and contract address of type Address
+
+    # create an RLN instance
+    var rlnInstance = createRLNInstance()
+    check: rlnInstance.isOk == true
+    # generate the membership keys
+    let membershipKeyPair = membershipKeyGen(rlnInstance.value)
+    check: membershipKeyPair.isSome
+    let pk = membershipKeyPair.get().idCommitment.toUInt256()
+    debug "membership commitment key", pk = pk
+
+    # test ------------------------------
+    var fut = newFuture[void]()
+    let s = await contractObj.subscribe(MemberRegistered, %*{"fromBlock": "0x0",
+        "address": contractAddress}) do(
+      pubkey: Uint256, index: Uint256){.raises: [Defect], gcsafe.}:
+      try:
+        debug "onRegister", pubkey = pubkey, index = index
+        check pubkey == pk
+        fut.complete()
+      except Exception as err:
+        # chronos still raises exceptions which inherit directly from Exception
+        doAssert false, err.msg
+    do (err: CatchableError):
+      echo "Error from subscription: ", err.msg
+
+    # register a member
+    let tx = await contractObj.register(pk).send(value = MEMBERSHIP_FEE)
+    debug "a member is registered", tx = tx
+
+    # wait for the event to be received
+    await fut
+
+    # release resources -----------------------
+    await web3.close()
+
+  asyncTest "insert a key to the membership contract":
+    # preparation ------------------------------
+    debug "ethereum client address", ETH_CLIENT
+    let contractAddress = await uploadRLNContract(ETH_CLIENT)
+    # connect to the eth client
+    let web3 = await newWeb3(ETH_CLIENT)
+    debug "web3 connected to", ETH_CLIENT
+
+    # fetch the list of registered accounts
+    let accounts = await web3.provider.eth_accounts()
+    web3.defaultAccount = accounts[1]
+    let add = web3.defaultAccount
+    debug "contract deployer account address ", add
+
+    # prepare a contract sender to interact with it
+    var sender = web3.contractSender(MembershipContract,
+        contractAddress) # creates a Sender object with a web3 field and contract address of type Address
+
+    # send takes three parameters, c: ContractCallBase, value = 0.u256, gas = 3000000'u64 gasPrice = 0
```

Comment:
`c: ContractCallBase` is one of the parameters? (if yes then are 4)
`value` should be `5.u256 ` instead of `0.u256`?

```suggestion
    # send takes three parameters, c: ContractCallBase, value = 0.u256, gas = 3000000'u64, gasPrice = 0
```

Reply:
> value should be 5.u256 instead of 0.u256?

0 is its default value but can be anything if the caller sets a different value

Reply:
Also edited the comment https://github.com/status-im/nwaku/pull/941/commits/be242a607c57eea03c9aa3027e0640406cc3a545